sql3/planner/inbuiltfunctionsstring.go

```diff
@@ -783,3 +784,85 @@ func (n *callPlanExpression) EvaluateCharIndex(currentRow []interface{}) (interf
 	}
 	return int64(res), nil
 }
+
+func (p *ExecutionPlanner) analyseFunctionStr(call *parser.Call, scope parser.Statement) (parser.Expr, error) {
+	if len(call.Args) < 1 || len(call.Args) > 3 {
+		return nil, sql3.NewErrCallParameterCountMismatch(call.Rparen.Line, call.Rparen.Column, call.Name.Name, 1, len(call.Args))
+	}
+	targetType := parser.NewDataTypeDecimal(4)
+	if !typesAreAssignmentCompatible(targetType, call.Args[0].DataType()) {
+		return nil, sql3.NewErrTypeAssignmentIncompatible(call.Args[0].Pos().Line, call.Args[0].Pos().Column, call.Args[0].DataType().TypeDescription(), targetType.TypeDescription())
+	}
+
+	// the second and third parameters are optional
+	for i := 1; i < len(call.Args); i++ {
+		if typeIsVoid(call.Args[i].DataType()) {
+			return nil, sql3.NewErrLiteralNullNotAllowed(call.Args[i].Pos().Line, call.Args[i].Pos().Column)
+		}
+		if !typeIsInteger(call.Args[i].DataType()) {
+			return nil, sql3.NewErrIntExpressionExpected(call.Args[i].Pos().Line, call.Args[i].Pos().Column)
+		}
+	}
+
+	call.ResultDataType = parser.NewDataTypeString()
+
+	return call, nil
+}
+
```

Comment:
Can you add documentation for this?

Reply:
Realized i was still connected to the old docs, moved it into a branch on the new docs
https://github.com/FeatureBaseDB/featurebase-docs/tree/fb-1862